synapse/models/inet.py

```diff
@@ -102,6 +103,41 @@ def norm(self, valu, oldval=None):
     def repr(self, valu):
         return valu.encode('utf8').decode('idna')
 
+class Rfc2822Addr(DataType):
+    '''
+    An RFC 2822 compatible email address parser
+
+    Specify strict=1 to strictly enforce complaince.
+    '''
+    def norm(self, valu, oldval=None):
+
+        # remove quotes for normalized version
+        valu = valu.replace('"', ' ').replace("'", ' ')
+        valu = valu.strip().lower()
+        valu = ' '.join(valu.split())
+
+        subs = {}
+
+        name, addr = email.utils.parseaddr(valu)
```

Comment:
If this parse goes boom do we want to raise a bad typevalue?

Reply:
I couldn't get it to go boom with an input string, but i can easily wrap it just in case.  thanks :)